test/ClientTest.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Rx\Websocket\Test;
+
+use React\EventLoop\Factory;
+
+class ClientTest extends \PHPUnit_Framework_TestCase
+{
+    public function testErrorBeforeRequest()
+    {
+        $loop = Factory::create();
+
+        $client = new \Rx\Websocket\Client('ws://127.0.0.1:12340/', false, [], $loop);
```

Comment:
No really, besides taking a below 0 or above 65535 port as those are in valid assuming that works